### ISSUE TITLE
Bug: Not recognising GitHub remote with credencials

### ISSUE THREAD

**Issue by tinovyatkin** [KnisterPeter/vscode-github#33]
The extension seems to not recognising GitHub remote with credentials inside like `https://user:pass@github.com/user/repo`

Error at DevTools:

```
//Applications/Visual Studio Code - Insiders.app/Contents/Resources/app/out/vs/workbench/electron-browser/workbench.main.js:130 Not a github project?: Error: Not a github project?
    at execa.then.result (/Users/tino/.vscode-insiders/extensions/KnisterPeter.vscode-github-0.7.0/out/src/git.js:17:23)
    at process._tickCallback (internal/process/next_tick.js:103:7)e.onUnexpectedError @ 
```

**Comment by KnisterPeter** [comment]
You are correct in this. Personally I would not add credentials to the URL but for anyone who want it should be possible.

**Comment by tinovyatkin** [comment]
It still correct URL, and it's a pain to not have it when you are working on Windows for example... I think it's better to use Node built-in `url` module to Parse http/https remotes instead of relying on RegEx in these cases

**Comment by KnisterPeter** [comment]
Pull request is greatly appreciated.

**Comment by tinovyatkin** [comment]
Will try to do that now, however I don´t have a lot of experience with Typescript, but seeing a way to import how you request `git remote` too..

**Comment by KnisterPeter** [comment]
Cool.

You can just import `import * as url from 'url'`. Node defaults are already typed.
Then just add the lines of code as you would with javascript.

If anything is unclear, don't hesistate to ask.

**Comment by KnisterPeter** [comment]
Closed with #38

**Comment by KnisterPeter** [comment]
Thanks again for contributing. :tada:

**Comment by KnisterPeter** [comment]
And published as 0.7.1
